### PR TITLE
Fix metrics unable to retrieve lazy inited executor status

### PR DESCRIPTION
**Pull request by AlbumenJ**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/store/support/SimpleDataStore.java [hunk 1]
```diff
@@ -60,5 +68,22 @@ public void remove(String componentName, String key) {
...
+    public void addListener(DataStoreUpdateListener dataStoreUpdateListener) {
+        listeners.add(dataStoreUpdateListener);
+    }
+
+    private void notifyListeners(String componentName, String key, Object value) {
+        for (DataStoreUpdateListener listener : listeners) {
+            try {
+                listener.onUpdate(componentName, key, value);
+            } catch (Throwable t) {
+                logger.warn(
```
Comment (songxiaosheng) [inline dubbo-common/src/main/java/org/apache/dubbo/common/store/support/SimpleDataStore.java:84]: Can the log print componentName, key, and value variables at the same time to facilitate troubleshooting?

File: dubbo-common/src/main/java/org/apache/dubbo/common/store/DataStore.java [hunk 2]
```diff
@@ -34,4 +34,7 @@ public interface DataStore {
     void put(String componentName, String key, Object value);
 
     void remove(String componentName, String key);
+
+    default void addListener(DataStoreUpdateListener dataStoreUpdateListener) {}
+    ;
```
Comment (songxiaosheng) [inline dubbo-common/src/main/java/org/apache/dubbo/common/store/DataStore.java]: Semicolon does not need to be wrapped

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java [hunk 3]
```diff
@@ -61,14 +62,28 @@ public ThreadPoolMetricsSampler(DefaultMetricsCollector collector) {
         this.collector = collector;
     }
 
+    @Override
+    public void onUpdate(String componentName, String key, Object value) {
+        if (EXECUTOR_SERVICE_COMPONENT_KEY.equals(componentName)) {
+            if (value instanceof ThreadPoolExecutor) {
+                addExecutors(SERVER_THREAD_POOL_NAME + "-" + key, (ThreadPoolExecutor) value);
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java]: `SERVER_THREAD_POOL_NAME + "-" +  `
前缀有多个地方用到了，需要提取到常量里面

**Review by songxiaosheng — CHANGES_REQUESTED** [review]

**Comment by AlbumenJ** [comment]
@CrazyHZM PTAL

**Review by CrazyHZM — APPROVED** [review]

**Pull request merged**